### DISCUSSION TITLE
Clarification on polar/cartesian coordinates

### DISCUSSION THREAD

**Discussion by jffaust** [thi-ng/umbrella#421] [Q&A]
Hello. I am using the cartersian3 function from https://github.com/thi-ng/umbrella/blob/develop/packages/vectors/src/cartesian.ts and I would like some help understanding its output. Based on the code, the provided polar vector should in the format of `[r,θ,ϕ]`. Trying it out, if I pass the following vector `[1,0,0]` (radius of 1 without any rotations), I receive a the following output: `[1,0,0]`. When I look online for spherical/polar coordinates system, the vector is aligned on the Z axis before the rotations are applied, like here on Wikipedia:

https://en.wikipedia.org/wiki/Spherical_coordinate_system#/media/File:3D_Spherical.svg

So I would expect the resulting vector to be `[0,0,1]`. Am I missing something where the result is not in the format of `[x,y,z]`?

**Comment by jffaust** [comment]
I do see the `(Z, X, Y)` order mentioned on Wikipedia so maybe its related to geographical directions? 
![image](https://github.com/thi-ng/umbrella/assets/4367222/a4aedd72-fc8b-42ef-a1be-07c04740500f)

**Comment by postspectacular** [comment]
Hi @jffaust - the implementation of `polar3()` & `cartesian3()` is just a port from my old Clojure library and I don't remember what code or convention it was based on (likely even on my earlier toxiclibs Java library from 2006). To be honest, I will have to check myself first, which of the conventions is used here — I find the Wikipedia entry semi-confusing, so need more time to compare before I can answer properly... Here's a quick REPL example for now, maybe you can make more sense yourself from these given cases/examples...

```js
new Map([[1,0,0], [0,1,0], [0,0,1], [-1,0,0], [0,-1,0], [0,0,-1], [1,1,0], [1,0,1], [0,1,1]].map(x => [x, polar3([], x)]))

// (manually replaced floats w/ multiples of π and sqrt2 for better legibility)
// Map(9) {
//   [ 1, 0, 0 ] => [ 1, 0, 0 ],
//   [ 0, 1, 0 ] => [ 1, 0, π/2 ],
//   [ 0, 0, 1 ] => [ 1, π/2, 0 ],
//   [ -1, 0, 0 ] => [ 1, 0, π ],
//   [ 0, -1, 0 ] => [ 1, 0, -π/2 ],
//   [ 0, 0, -1 ] => [ 1, -π/2, 0 ],
//   [ 1, 1, 0 ] => [ sqrt2, 0, π/4 ],
//   [ 1, 0, 1 ] => [ sqrt2, π/4, 0 ],
//   [ 0, 1, 1 ] => [ sqrt2, π/4, π/2 ]
// }
```

Ps. I do know the roundtrip conversion between the both functions works and that has usually been enough for me. I only just found (and fixed) a bug in `polar3()` for 0-vectors...

**Reply by jffaust** [reply]
Thanks for the response. I'll try to figure things out but I'm also considering pausing this particular project for something a bit simpler (always dreaming too big ahah).

You can be sure though that the other projects are all using thi.ng and I'm really grateful for all the functionality it provides. There's still so much stuff to discover.

I'll gladly share my work when I'm finally able to complete my first project.
